dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/ModuleSelectionConfigImpl.java

```diff
@@ -173,16 +243,55 @@ public List<String> findIncludesNotMatchingAnything(Collection<? extends ModuleI
             }
         }
 
-        return includedNotYetSeen.stream().map(ModuleName::toString).collect(toList());
+        List<ModuleName> includedNotYetSeenButSeenUnderOldNaming = new ArrayList<>();
+        if (serverxmlmode && ProductInfo.getBetaEdition()) {
+            for (Iterator<ModuleName> iterator = includedNotYetSeen.iterator(); iterator.hasNext();) {
+                ModuleName moduleName = iterator.next();
+                for (ModuleInfo moduleInfo : moduleInfos) {
+                    if (matches(moduleName, moduleInfo, false)) {
+                        includedNotYetSeenButSeenUnderOldNaming.add(moduleName);
+                        iterator.remove();
+                        break;
+                    }
+                }
+            }
+
+            for (String unmatchedInclude : includedNotYetSeenButSeenUnderOldNaming.stream().map(ModuleName::toString).collect(toList())) {
+                Tr.warning(tc, MessageConstants.OPENAPI_USING_WRONG_NAME_SOURCE_CWWKO1680W, Constants.MERGE_INCLUDE_CONFIG, unmatchedInclude);
+            }
+        }
```

Comment:
Can we do this check that the may have used the wrong name for excluded apps as well as included apps please?

(The other check for a name that's missing should remain as only checking the included apps).